internal/collector/telemetry/telemetry.go

```diff
@@ -25,11 +25,14 @@ const (
 )
 
 var (
+	metricsAddrDefault   = ":8888"
+	metricsPrefixDefault = "otelcol"
+	addInstanceIDDefault = true
 	// Command-line flags that control publication of telemetry data.
-	metricsAddrPtr   *string
-	metricsPrefixPtr *string
+	metricsAddrPtr   = &metricsAddrDefault
+	metricsPrefixPtr = &metricsPrefixDefault
```

Comment:
```suggestion
	metricsAddrPtr   = new(string)
	metricsPrefixPtr = new(string)
```

Would this be safer?  Does anything use the default value that might break if it changes when the user specifies this value via CLI flags?

Reply:
On this particular instance it would be fine to just use `new`, but on the logs side it's not https://github.com/open-telemetry/opentelemetry-collector/blob/afad36e3785b78906f6eb5f9e91354a9e05a274c/service/internal/telemetrylogs/logger.go#L78-L90 since you need the value to be `"deprecated"`. I would prefer to leave it like this for consistency